### ISSUE TITLE
MacPassHTTP has high response time on MacPass 0.7.1 (17630)

### ISSUE THREAD

**Issue by mbrucco** [MacPass/MacPassHTTP#47]
Hi,
yesterday I upgraded MacPass to new release, but now integration from "ChromeIPass" on Chrome and "PassIFox" on FireFox are not working...

I deleted "Shared enchription keys" and "Stored permission", I connected browser extensions again, but they are not working.

Please, any idea to solve it?

BR
Maurizio

**Comment by mstarke** [comment]
Have you updated MacPassHTTP to version 0.3?

**Comment by mbrucco** [comment]
Yes I did it. It is not working... no entries returned and filled.

**Comment by mstarke** [comment]
That is strange, since others report it works.

**Comment by mstarke** [comment]
Can you check any logs/output from MacPass or the extensions on what they try to do and what fails?

- Does the MacPassHTTP Webserver work? (try to connect via ```telnet``` for example
- Are port settings correct? (take a look at the settings)
- Can you manually delete the ```KeePassHTTP Settings``` entry in the root group?

**Comment by mbrucco** [comment]
Sorry, I tried to locate logs.. but without success.

Can you please help me to locate them?

I tried to reinstall both without positive results.

**Comment by mbacodes** [comment]
From Update Notice of MacPass:

> Version 0.7.x breaks the old plugin API. Thus you need to update any plugins to work again!

**Comment by mstarke** [comment]
@mbacodes @mbrucco stated that he has updated to 0.3.

@mbrucco: Is MacPassHTTP displayed in the plugin settings of MacPass? Did you try to telnet into the server?

**Comment by mbrucco** [comment]
Hi,
I try to attach print screen


I have 0.3 version installed and visible…

Leaving a web page open on login form, I noticed that after a very huge amount of minutes a notification appears… so now… it works… but too much slow….

How can be fixed the incredible delay?

BR
Maurizio

**Comment by mstarke** [comment]
Can you post some statistics on you database? Is the file very large? Can you test the setup with a clean and simple example database?

The plugin is not my main focus - MacPass is so I'm very open to any suggestions on how to debug this.

**Comment by mbrucco** [comment]
Thank a lot!!!! Your idea is good… issue is related to number of password stored, I tested with single entry and it work immediately


But my problem now is… how can I manage it? I have about 300 entries. Main reason is impossibility to add url alternatives, so I’ve to replicate entries for each possible url for same credentials.

I think something changed in how application reply to plug-in.

**Comment by mstarke** [comment]
I ran a small test to see what might have caused the change. Do you use a simple entries or use those entries a lot of references and/or placeholders?

**Comment by mstarke** [comment]
And did you consider just using Autotype with window associations? This might match multiple URLs better but you lose some of the comfort of the plugin.

**Comment by mbrucco** [comment]
Yes I know.. but plug-in is a must for me…

**Comment by mstarke** [comment]
Again, do you use References or plain copies of the entries?

I just ran a test with a database with 256 entries with the same url added in 16 nested groups and the response time was instant. No reference or placeholders though. This might be the bottleneck since reference resolving was changed since the last release and is not really optimised for speed.

**Comment by mbrucco** [comment]
I use a lot of reference 

Inviato da iPhone

**Comment by mstarke** [comment]
I changed the setup with a database with 256 entries all with references to the same entry but the request is still very fast. I cannot get the low speed responses. Is the database file located anywhere special? Can you try to create a simple test database that has the same issues so I can investigate any further?

**Comment by mbrucco** [comment]
Ok, thanks.

Now I’m trying to prepare test database, it requires a lot time for me… as soon as I have it I’ll share with you.

Thanks again
Maurizio

**Comment by mstarke** [comment]
Sorry for the inconvenience but I highly appreciate the effort!

**Comment by mstarke** [comment]
Any progress on your end?

**Comment by mbrucco** [comment]
HI,
Unfortunately no… I started getting my actual db, but I realised there are too much entries to clean password… but I discovered that I’ve to clean history too…

Is there any alternatives to perform it manually?

BR
Maurizio

**Comment by mstarke** [comment]
<del>You can disable the history and MacPass will clean all the history of all entries. Another way is to save the file as .kdb since kdb does not support history but you will lose other things too that might contribute to your long response times.</del> Forget that, do not try to clean up your DB. We should focus on creating a dummy on that mirrors your setup.

I can create that dummy DB for you and you can try to use this one and see what response times you get.

You said you use a lot of references - what kind of reference are you using? UUID based or do you query other fields of entries? This way I might be able to create a setup that represents yours and can determine where the time gets used up the most.

**Comment by mbrucco** [comment]
Ho. Thanks.

I use uuid ref for user and password 

Maurizio 

Inviato da iPhone

**Comment by mbrucco** [comment]
Hi, had you opportunity to complete dummy db?
